src/DssBlow2.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-FileCopyrightText: © 2023 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+pragma solidity ^0.8.24;
+
+interface ERC20Like {
+    function balanceOf(address) external returns (uint256);
+    function approve(address usr, uint256 wad) external returns (bool);
+}
+
+interface JoinLike {
+    function dai() external view returns (address);
+    function usds() external view returns (address);
+    function join(address, uint256) external;
+}
```

Comment:
This is a bit problematic because `DaiJoin` doesn't define a `usds()` function.
You might want to split the interfaces, even if that leads to some duplication or requires some inheritance:

```suggestion
interface DaiJoinLike {
    function dai() external view returns (address);
    function join(address, uint256) external;
}

interface UsdsJoinLike is DaiJoinLike {
    function usds() external view returns (address);
}
```